src/node/nodestate.h

```diff
@@ -278,13 +272,21 @@ namespace ccf
 
           self = 0; // The first node id is always 0
 
-#ifdef PBFT
-          setup_pbft(args.config);
-#else
-          setup_raft();
-#endif
+          if (args.consensus_type == ConsensusType::Pbft)
```

Comment:
This should be moved to its own `setup_consensus()` function as it is duplicated in the join handler as well.